core/src/main/scala/com/qvantel/jsonapi/ApiEndpoint.scala

```diff
@@ -0,0 +1,17 @@
+package com.qvantel.jsonapi
+
+import cats.effect.IO
+import com.netaporter.uri.Uri
+
+trait ApiEndpoint {
+  def uri: IO[Uri]
+}
+
+object ApiEndpoint {
+  final case class Static(static: Uri) extends ApiEndpoint {
+    override val uri: IO[Uri] = IO.pure(static)
+  }
+  final class Dynamic(f: () => IO[Uri]) extends ApiEndpoint {
```

Comment:
Shouldn't this be just `f: IO[Uri]`?

Reply:
Or even: `final case class Dynamic(uri: IO[Uri])`

Reply:
Changed to a case class.